### PR TITLE
New yeoda filename

### PR DESCRIPTION
**Pull request by fl0roth**
Implementation of the new filename and folder logic with a more compact and general purpose use. The corresponding changes include:
- Introduction of a yeoda_naming class which represents the new filenaming and folder logic and a corresponding test_yeoda_naming routine. 
- Introduction of fields with variable length (defined by length = 0) in the SmartFilename and SmartFilenamePart class.
- Introduction of an option (input parameter "compact") to use a compact filename design. If it is set to True, empty fields are represented by a single pad character and all fields are treated with variable length.  
- Adaption of the attached conda environment by updating the python-dateuntil modul (issue#18).

### CODE REVIEW

**Review by bbauerma — APPROVED** [review]
Looks fine for me.
a) yeoda-filename works as expected
b) all tests on my local Windows machine are green

**Review by FelixReuss — APPROVED** [review]
- works fine with S1 Sigma

File: src/geopathfinder/file_naming.py [hunk 1]
```diff
@@ -334,10 +367,14 @@ def _build_fn(self):
...
         """
+
         fn_parts = list(self._fn_map.values())
         filename = str(fn_parts[0])
         for i in range(1, len(fn_parts)):
-            filename = filename[:-len(fn_parts[i-1])] + (fn_parts[i-1] + fn_parts[i])
+            if len(fn_parts[i-1]) == 0:
+                filename = filename + (fn_parts[i - 1] + fn_parts[i])
+            else:
+                filename = filename[:-len(fn_parts[i-1])] + (fn_parts[i-1] + fn_parts[i])
```
Comment (cnavacch) [inline src/geopathfinder/file_naming.py:377]: That part can be made shorter, I will fix this, because this was originally implemented by myself.

**Review by cnavacch — CHANGES_REQUESTED** [review]
Looks very good! Three changes still need to be done:

1. Fixing Python version (@bbauerma @fl0roth ?)
2. Change loop in `_build_fn()` (I will do this).
3. Add `compact=False` in the `__init__` of all other file naming conventions. This might me a nice feature to have for the existing file naming conventions (@fl0roth ).

File: conda_env.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ name: geopathfinder
 channels:
   - defaults
 dependencies:
-  - python=3.6
+  - python
```
[the remark this answers is not in the document]
Comment (bbauerma) [reply conda_env.yml:5]: actually, can we remove python at all?

[the remark this answers is not in the document]
Comment (bbauerma) [reply conda_env.yml:5]: it should work for python 3. for all python 3 versions (3.6, 3.7,..)

[the remark this answers is not in the document]
Comment (sebhahn) [reply conda_env.yml:5]: I would recommend >= 3.6

[the remark this answers is not in the document]
Comment (cnavacch) [reply conda_env.yml]: The issue I see at the moment is that we use the same conda yml file for Travis. I prefer to have separate yml files for testing each Python version with Travis. This ensures that each Python version works/is supported and is always tested and makes it easier to reset package dependencies for a specific Python version. In the main yml file I agree with @sebhahn , there we could set `python >= 3.6`.

[the remark this answers is not in the document]
Comment (bbauerma) [reply conda_env.yml]: is already set to python >= 3.6 now
please proceed and implement the steps you suggested - i want to release a new version...

**Review by cnavacch — APPROVED** [review]
Great, all changes are fine from my side!

**Pull request merged**